src/vaults/AffineVault.sol

```diff
@@ -424,33 +432,75 @@ contract AffineVault is AffineGovernable, AccessControlUpgradeable {
-     * @param amount The amount we want to liquidate
-     * @return The amount we actually liquidated
+     * @param assetsRequested The amount we want to liquidate
+     * @return assetsLiquidated debtCreated
      */
-    function _liquidate(uint256 amount) internal returns (uint256) {
-        uint256 amountLiquidated;
+    function _liquidate(uint256 assetsRequested) internal returns (uint256 assetsLiquidated, uint256 debtCreated) {
+        uint256 debtStrategyBps;
+        DivestResponse[MAX_STRATEGIES] memory responses;
+
         for (uint256 i = 0; i < MAX_STRATEGIES; i = uncheckedInc(i)) {
             Strategy strategy = withdrawalQueue[i];
-            if (address(strategy) == address(0)) {
-                break;
-            }
+            if (address(strategy) == address(0)) break;
 
             uint256 balance = _asset.balanceOf(address(this));
-            if (balance >= amount) {
-                break;
+            if (balance >= assetsRequested) break;
+
+            // Withdraw `asset`
+            (uint256 assetsFromStrat, DivestResponse response) =
+                _withdrawFromStrategy(strategy, assetsRequested - balance, DivestType.POSSIBLE);
+            assetsLiquidated += assetsFromStrat;
+
+            if (response == DivestResponse.DEBT) {
+                debtStrategyBps += strategies[strategy].tvlBps;
+                responses[i] = DivestResponse.DEBT;
             }
+        }
+
+        emit Liquidation({assetsRequested: assetsRequested, assetsLiquidated: assetsLiquidated});
 
-            uint256 amountNeeded = amount - balance;
-            amountNeeded = Math.min(amountNeeded, strategies[strategy].balance);
+        uint256 currAssets = _asset.balanceOf(address(this));
 
-            // Force withdraw of token from strategy
-            uint256 withdrawn = _withdrawFromStrategy(strategy, amountNeeded);
-            amountLiquidated += withdrawn;
+        // If we liquidated enough OR there are no strategies with debt, exit
+        if (assetsRequested <= currAssets || debtStrategyBps == 0) {
+            return (assetsLiquidated, debtCreated);
         }
-        emit Liquidation({assetsRequested: amount, assetsLiquidated: amountLiquidated});
-        return amountLiquidated;
+
+        // Issue debt to strategies which failed to liquidate. Debt is issued proportionally according to assigned tvl bps
+        uint256 debtToCreate = assetsRequested - currAssets;
+        debtCreated = debtToCreate;
+        for (uint256 i = 0; i < MAX_STRATEGIES; i = uncheckedInc(i)) {
+            Strategy strategy = withdrawalQueue[i];
+            if (address(strategy) == address(0)) break;
+            if (responses[i] != DivestResponse.DEBT) continue;
+
+            uint16 stratBps = strategies[strategy].tvlBps;
+            uint256 debt = (debtToCreate * stratBps) / debtStrategyBps;
+            debtToCreate -= debt;
+            debtStrategyBps -= stratBps;
+            strategy.increaseDebt(debt);
```

Comment:
Assigning debt to all strategy after doing possible withdrawal might end up empty strategy  having a debt. We should only assign debt to strategy which fails to withdraw or withdraw less than expected.